### PR TITLE
feat(Bonus Pagamenti Digitali): [#176455937] Error and loading screen transaction details

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR add the loading and the error screen in the section "transaction details" of a bpd period detail.

## List of changes proposed in this pull request
- Added the TransactionUnavailable screen
- Added the LoadTransactions screen
- Create a proxy component to manage the different screens

**Available transaction**

**Loading transaction**

**Unavailable transaction**

### CODE REVIEW

**Review by CrisTofani — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/screens/details/BpdDetailsScreen.tsx [hunk 1]
```diff
@@ -121,13 +127,14 @@ const BpdDetailsScreen: React.FunctionComponent<Props> = props => {
...
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  load: () => dispatch(bpdAllData.request()),
   completeUnsubscription: () => {
     dispatch(bpdAllData.request());
     dispatch(bpdUnsubscribeCompleted());
     dispatch(NavigationActions.back());
   },
-  goToTransactions: () => dispatch(navigateToBpdTransactions()),
+  goToTransactions: () => {
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/features/bonus/bpd/screens/details/BpdDetailsScreen.tsx]: Sure, fixed in [01eb23e](https://github.com/pagopa/io-app/pull/2755/commits/01eb23e3f5fddf6f3f4440d2a1cb68d6713c6fcd)

File: ts/features/bonus/bpd/screens/details/transaction/TransactionsUnavailable.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+>;
+
+const loadLocales = () => ({
+  headerTitle: I18n.t("bonus.bpd.details.transaction.goToButton"),
+  title: I18n.t("bonus.bpd.details.transaction.error.title"),
+  body: I18n.t("bonus.bpd.details.transaction.error.body")
+});
+
+/**
+ * This screen informs the user that there are problem in retrieve the transactions list.
```
Comment (fabriziofff) [inline ts/features/bonus/bpd/screens/details/transaction/TransactionsUnavailable.tsx:22]:
```suggestion
 * This screen informs the user that there are problems retrieving the transactions list.
```

**Review by fabriziofff — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/screens/details/BpdDetailsScreen.tsx [hunk 3]
```diff
@@ -104,7 +104,13 @@ const BpdDetailsScreen: React.FunctionComponent<Props> = props => {
...
         footerContent={
           canRenderButton && (
-            <GoToTransactions goToTransactions={props.goToTransactions} />
+            <GoToTransactions
+              goToTransactions={() =>
+                // if the transactions state is in pot.isError state
+                // on press request a refresh
+                props.goToTransactions()
+              }
+            />
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/features/bonus/bpd/screens/details/BpdDetailsScreen.tsx]: You're right! Fixed in [f05e33e](https://github.com/pagopa/io-app/pull/2755/commits/f05e33e6d60f5d8c4e7c37339504c0e7fa581ec5)

File: ts/features/bonus/bpd/screens/details/transaction/TransactionsUnavailable.tsx (hunk 2 above)
Comment (debiff) [reply ts/features/bonus/bpd/screens/details/transaction/TransactionsUnavailable.tsx]: Fixed in [156a52a](https://github.com/pagopa/io-app/pull/2755/commits/156a52ae7391c759b440fc09100092134189e1b1)

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**